### ISSUE TITLE
ToDo app : Ready for review

### ISSUE THREAD

**Issue by vrumale** [vrumale/SimpleToDo#1]
Simple ToDo app is complete. Please review the app. @codepathreview and @codepath

I plan to add more features as suggested as time permits.

**Comment by nesquena** [comment]
Vinutha,

Looks good! This was intended in part to give you an introduction to the general rhythm of this course. The course is entirely project based with an app being assigned each week and then due the following week. Each project builds on the last to help each engineer to learn all practical Android development and best practices as quickly as possible. We also do a personalized code review for each submitted project once the bootcamp starts.

The next step is to continue working on [extensions to your todo app](http://courses.codepath.com/snippets/intro_to_android/prework#heading-5-extending-your-todo) and to **schedule a short 5-10 minute phone conversation** [here](https://www.google.com/calendar/selfsched?sstoken=UURvT01vbzBza1pnfGRlZmF1bHR8NzVjYzBkN2EyZTRiOGRmYjZlZjhmMjA2N2IyMGMwYWU). Navigate to [January 8th and dates onward](https://www.google.com/calendar/selfsched?sstoken=UURvT01vbzBza1pnfGRlZmF1bHR8NzVjYzBkN2EyZTRiOGRmYjZlZjhmMjA2N2IyMGMwYWU) and choose a 15-minute slot. Let us know if none of those times work.

Please make sure to **include the best number** to reach you at in the scheduled event? Look forward to chatting soon!

**Comment by vrumale** [comment]
Hi Nathan, 

Thanks for the quick review. I has adding my number and scheduled a call on your calender. Looking forward to talking to you.

**Comment by nesquena** [comment]
Great, chat with you tomorrow!

**Comment by nesquena** [comment]
Just wanted to check in as a reminder. Starting today will begin accepting engineers that complete 1-2 of the [suggested extensions](http://courses.codepath.com/snippets/intro_to_android/prework#heading-5-extending-your-todo) within the next week. Happy new year!

**Comment by vrumale** [comment]
Hi Nathan,

I'm trying to use SQLite to add todo items. I had a question If I device my
own ToDoItem objects and try to feed this to the ArrayAdapter do I need to
define my own Custom adapter?

Thanks,
Vinutha

**Comment by vrumale** [comment]
Added SQLite support @codepathreview @codepath Please review

**Comment by nesquena** [comment]
Cool good to see that. However, there's some room for improvement on how you take the items from SQL to your adapter. For example you could do an `ArrayAdapter<TodoItem>` and then implement the `toString` method in TodoItem model for the same effect without having to copy all the strings separately. Ultimately in the next extension you should work on one that allows you to create a custom adapter anyways. Ping me again once you've submitted that.

**Comment by vrumale** [comment]
Hey Nathan, 

I sent you an email about that not sure if you saw it. I tried to do `ArrayAdapter<TodoItem>`and could not proceed. I thought with that approach I will need a Custom adapter.

**Comment by nesquena** [comment]
Basically need to change from `String` to the java object `TodoItem` in the adapter types and then setup the `toString` method on the java object to have it return the "body". But you should play around with a custom adapter anyhow for the next extension.

**Comment by vrumale** [comment]
The part I could not get was how to set only the String to the ListView and ignore the other parts since the `ArrayList` is of type TodoItem

**Comment by nesquena** [comment]
Yeah unless you know about overriding the `toString` method on the class, it would be hard to do.

**Comment by vrumale** [comment]
Is this something the Custom Adapter would resolve? Would it help in fine tuning how a Java Object can be mapped to ListView?

**Comment by nesquena** [comment]
That's it's exact purpose to allow you to render the object to a list item exactly how you want as [outlined here](https://guides.codepath.com/android/Using-an-ArrayAdapter-with-ListView#using-a-custom-arrayadapter)

**Comment by vrumale** [comment]
Made changes to override toString() Please review @codepathreview @codepath

**Comment by nesquena** [comment]
Cool, that definitely cleaned it up a bit. I would encourage you to work on one more extension and a custom adapter as I mentioned in earlier message. Will keep an eye out for any other extensions you submit.

**Comment by vrumale** [comment]
Adding Custom Adapter @codepathreview @codepath @nesquena

**Comment by nesquena** [comment]
Great, nice to see you got in the custom adapter as well as the SQLite support now! I will be in touch again soon via email.
